src/bundles/csg/functions.ts

```diff
@@ -0,0 +1,823 @@
+      );
+    } else {
+      return bounds[j][i];
+    }
+  };
+}
+
+/**
+ * Rotate the shape by the provided angles in the x, y and z direction.
+ * Angles provided are in the form of radians (i.e. 2π represent 360
+ * degrees)
+ *
+ * @param {Shape} shape - The shape to be rotated
+ * @param {number} x - Angle of rotation in the x direction
+ * @param {number} y - Angle of rotation in the y direction
+ * @param {number} z - Angle of rotation in the z direction
+ * @returns {Shape} The rotated shape
+ */
+export function rotate(shape: Shape, x: number, y: number, z: number): Shape {
+  let newSolid: Solid = _rotate([x, y, z], shape.solid);
+  return new Shape(newSolid);
+}
+
+/**
+ * Rotate the shape by the provided angles in the x direction. Angles
+ * provided are in the form of radians (i.e. 2π represent 360 degrees)
+ *
+ * @param {Shape} shape - The shape to be rotated
+ * @param {number} x - Angle of rotation in the x direction
+ * @returns {Shape} The rotated shape
+ */
+export function rotate_x(shape: Shape, x: number): Shape {
+  return rotate(shape, x, 0, 0);
+}
+
+/**
+ * Rotate the shape by the provided angles in the y direction. Angles
+ * provided are in the form of radians (i.e. 2π represent 360 degrees)
+ *
+ * @param {Shape} shape - The shape to be rotated
+ * @param {number} y - Angle of rotation in the y direction
+ * @returns {Shape} The rotated shape
+ */
+export function rotate_y(shape: Shape, y: number): Shape {
+  return rotate(shape, 0, y, 0);
+}
+
+/**
+ * Rotate the shape by the provided angles in the z direction. Angles
+ * provided are in the form of radians (i.e. 2π represent 360 degrees)
+ *
+ * @param {Shape} shape - The shape to be rotated
+ * @param {number} z - Angle of rotation in the z direction
+ * @returns {Shape} The rotated shape
+ */
+export function rotate_z(shape: Shape, z: number): Shape {
+  return rotate(shape, 0, 0, z);
+}
+
+//TODO
+/**
+ * Center the provided shape with the middle base of the shape at (0, 0, 0).
+ *
+ * @param {Shape} shape - The shape to be centered
+ * @returns {Shape} The shape that is centered
+ */
+function shapeSetOrigin(shape: Shape) {
+  let newSolid: Solid = align({ modes: ['min', 'min', 'min'] }, shape.solid);
+  return new Shape(newSolid);
+}
+
+/**
+ * Checks if the specified argument is a Shape.
+ *
+ * @param {unknown} argument - The value to check.
+ * @returns {boolean} Whether the argument is a Shape.
+ */
+export function is_shape(argument: unknown): boolean {
+  return argument instanceof Shape;
+}
+
+/**
+ * Creates a clone of the specified Shape.
+ *
+ * @param {Shape} shape - The Shape to be cloned.
+ * @returns {Shape} The cloned Shape.
+ */
+export function clone(shape: Shape): Shape {
+  return shape.clone();
+}
+
+/**
+ * Stores a clone of the specified Shape for later rendering.
+ *
+ * @param {Shape} shape - The Shape to be stored.
+ */
+export function store(shape: Shape): void {
```

Comment:
We rely on the `render...` group of functions to provide REPL feedback, eg `<Render #2>`, should the program as a whole evaluate to their return values. Storing does not return an undocumented dummy value serving only to provide a `.toReplString()` (we also do not wish to return the cloned or coloured cloned shape that we expect to no longer be modified), and it may not be a good idea to separately display additional logs in the REPL.

Storing a non-shape would (eventually) result in an error, especially when it comes time to render. Improved param validation would allow for storing to always work in theory (checking for shape, continued evaluation meaning it worked) and for the program to immediately error otherwise